### PR TITLE
Fix torch.triu / torch.tril on contiguous tensors with non-default st…

### LINKED ISSUE #22581: Batched Triu And Tril Incorrect for Some Inputs
## 🐛 Bug

When passing a specific 4D tensor to `triu` or `tril`, the GPU implementation produces non-deterministic results. The CPU implementation produces `nan` values.

## To Reproduce

This is the minimum reproducible example I could come up with:

```python
x = torch.randn(1, 4, 4, 4)
x = x.transpose(0, 1)
for i in range(10):
    # note: results are often different on each run
    # or on CPU, outputs `nan`
    print(x.triu().sum())
```

Note that the issue does not seem to occur when:
- the transpose is omitted
- if `x.size(0) > 1`
- if the tensor has more or fewer dimensions

## Expected behavior

Values should be deterministic and not produces `NaN`

## Environment

PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: Tesla V100-SXM2-16GB
Nvidia driver version: 418.67
cuDNN version: 7501

Versions of relevant libraries:
[pip] numpy==1.16.1
[pip] torch==1.1.0
[pip] torchfile==0.1.0
[conda] torch                     1.1.0                     <pip>
[conda] torchfile                 0.1.0                     <pip>

### PR DESCRIPTION
**Pull request by vishwakftw**
…rides

Changelog:
- Fix behavior of `torch.triu` / `torch.tril` on certain unsqueezed tensors that lead to uninitialized values on CPU

Test plan:
- Add tests for these cases in test_triu_tril in test_torch

Fixes #22581

### CODE REVIEW

File: aten/src/ATen/native/LinearAlgebraUtils.h [hunk 1]
```diff
@@ -48,22 +48,33 @@ static inline int64_t matrixStride(const Tensor& batched_matrices) {
  * Here batch contiguity is checked for tensors with greater than 4 dimensions.
  * Contiguous tensors and tensors with less than 3 dimensions pass this check
  */ 
-static inline bool checkTrilTriuBatchContiguous(const Tensor& tensor) {
+static inline std::tuple<bool, Tensor> checkTrilTriuBatchContiguous(const Tensor& tensor) {
```
Comment (zou3519) [inline aten/src/ATen/native/LinearAlgebraUtils.h]: nit: Returning two arguments like this is confusing to read (at least for me), but I'm not sure how to improve it.

Maybe a signature like
```
Tensor coerceTrilTriuBatchContiguous(const Tensor& tensor, bool* was_already_batch_contiguous)
```
might be clearer.

**Review by zou3519 — APPROVED** [review]

File: aten/src/ATen/native/LinearAlgebraUtils.h (hunk 1 above)
Comment (vishwakftw) [reply aten/src/ATen/native/LinearAlgebraUtils.h]: I thought about keeping the code more descriptive, and hence didn't consider a LAPACK-like return strategy.

But if you think that it reads better, I can make a change like below:
```c++
Tensor coerceTrilTriuBatchContiguous(const Tensor& tensor, bool& was_already_batched) {
  ...
}
```

Comment (zou3519) [reply aten/src/ATen/native/LinearAlgebraUtils.h]: Let's just keep what you have for now

Comment (vishwakftw) [reply aten/src/ATen/native/LinearAlgebraUtils.h]: Sure, thanks for the review.

**Pull request closed without merging**